gravitee-apim-console-webui/src/portal/homepage/homepage.component.ts

```diff
@@ -66,19 +77,56 @@ export class HomepageComponent {
       .subscribe();
 
     effect(() => {
-      this.contentControl.reset(this.portalHomepage().content);
+      this.contentControl.reset(this.portalHomepage()?.content || '');
     });
 
     effect(() => {
-      if (this.canUpdate()) {
-        this.contentControl.enable();
-      }
+      this.canUpdate() ? this.contentControl.enable() : this.contentControl.disable();
     });
   }
 
+  public togglePublish(): void {
+    const isCurrentlyPublished = this.portalHomepage()?.published;
+    const pageId = this.portalHomepage()?.id;
+
+    const data: GioConfirmDialogData = {
+      title: `${this.togglePublishActionText()} page?`,
+      content: isCurrentlyPublished
+        ? `This action will unpublish the page. It will no longer be visible on the developer portal, but you can publish it again at any time.`
+        : `Your changes will be published. The updated page will be visible on your developer portal.`,
+      confirmButton: this.togglePublishActionText(),
+    };
+
+    const toggleApiCall$ = isCurrentlyPublished
+      ? this.portalPagesService.unpublishPage(pageId)
+      : this.portalPagesService.publishPage(pageId);
+
+    this.matDialog
+      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
+        data,
```

Comment:
```suggestion
      .open<GioConfirmDialogComponent, GioConfirmDialogData, boolean>(GioConfirmDialogComponent, {
        width: GIO_DIALOG_WIDTH.SMALL,
        data,
```

i missed it in my first review but to fit with the figma design I would have added this. Wdyt ?

<img width="1728" height="876" alt="image" src="https://github.com/user-attachments/assets/b3e9e776-a3af-4bc0-8cff-ebe5c3bde33c" />

Reply:
added